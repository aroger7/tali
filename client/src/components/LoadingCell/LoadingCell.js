import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import ContentLoader from 'react-content-loader';

import useWindowSize from 'hooks/useWindowSize';
import { colors } from 'styles';

const LoadingCell = ({ value, isLoading }) => {
  const [cellWidth, setCellWidth] = useState(100);
  const windowSize = useWindowSize();
  const svgRef = useRef();

  useEffect(() => {
    if (svgRef && svgRef.current) {
      setCellWidth(svgRef.current.clientWidth);
    }
  }, [windowSize])

  return isLoading 
    ? (
      <div ref={svgRef}>
        <ContentLoader
          viewBox={`0 0 ${cellWidth} 13`}
          speed={1.2} 
          backgroundColor={colors.almostWhite}
          backgroundOpacity="0.3"
          foregroundColor="#dedede"
          foregroundOpacity="0.3"
          width="100%"
          height={13}
        >
          <rect x="0" y="0" rx="5" ry="5" width={cellWidth} height="13" />
        </ContentLoader>
      </div>
    )
    : <div>{value}</div>
};

LoadingCell.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node
  ]),
  isLoading: PropTypes.bool
};

export default LoadingCell;
