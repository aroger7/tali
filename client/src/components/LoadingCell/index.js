export { default } from './LoadingCell';
export * from './LoadingCell';
