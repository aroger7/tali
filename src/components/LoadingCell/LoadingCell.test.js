import { render, screen } from '@testing-library/react';
import React from 'react';
import LoadingCell from './LoadingCell';

describe('LoadingCell', () => {
    const value = 'test';
    const defaultProps = { value };

    it('should render a value', () => {
        const { asFragment } = render(<LoadingCell {...defaultProps} />)

        expect(asFragment()).toMatchSnapshot();
        expect(screen.getByText(value)).toBeInTheDocument();
    })

    it('should render a loading image', () => {
        render(<LoadingCell {...defaultProps} isLoading />);

        expect(screen.getByRole('img')).toBeInTheDocument();
    });
});
