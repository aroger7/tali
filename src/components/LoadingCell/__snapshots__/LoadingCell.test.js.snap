// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCell should render a value 1`] = `
<DocumentFragment>
  <div>
    test
  </div>
</DocumentFragment>
`;
